sphinx/ext/inheritance_diagram.py

```diff
@@ -378,15 +378,15 @@ def run(self) -> list[Node]:
                 aliases=self.config.inheritance_alias,
                 top_classes=node['top-classes'])
         except InheritanceException as err:
-            return [node.document.reporter.warning(err, line=self.lineno)]  # type: ignore[union-attr]
+            return [node.document.reporter.warning(err, line=self.lineno)]
 
         # Create xref nodes for each target of the graph's image map and
         # add them to the doc tree so that Sphinx can resolve the
         # references to real URLs later.  These nodes will eventually be
         # removed from the doctree after we're done with them.
         for name in graph.get_all_class_names():
             refnodes, x = class_role(  # type: ignore[call-arg,misc]
-                'class', ':class:`%s`' % name, name, 0, self.state)
+                'class', ':class:`%s`' % name, name, 0, self.state.inliner)
```

Comment:
Here, `class_role` is an object implementing the RoleFunction-like callable, namely

```
RoleFunction = Callable[
    [str, str, str, int, Inliner, dict[str, Any], Sequence[str]],
    tuple[list[nodes.Node], list[nodes.system_message]],
]
```

Stated otherwise, you can think of `class_role` as 

```
class_role(role_name, rawtext, text, lineno, inliner, options, content)
```

with *options* and *content* optional. So yes, we do expect an inliner, but if you want to be sure whether the call is correct or not, check the runtime types directly. But I think the call was incorrect. However, the `inliner` is likely not used, and that's why we never detected that issue. You'll may be able to remove some `# type: ignore` as well.

Reply:
Ok; these changes have been merged, but I _have not_ yet confirmed this at runtime, so we should do that to avoid unexpected side-effects.

Reply:
Ok; I've now checked the runtime behaviour, and the recipient of this function call is indeed a `role_adapter` function that accepts an `inliner: Inliner` as expected.

https://github.com/sphinx-doc/sphinx/blob/e63e2f3ccbe73f6f5b7608de455a12005c9c2865/sphinx/domains/__init__.py#L266-L271

So I'm comfortable with this change.